src/polyfills/formdata-entries.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Creates a new `FormData` object.
+ *
+ * This patches the native constructor so that the polyfill can capture the
+ * fields in the form passed to the constructor. If `FormData#entries` needs a
+ * polyfill, chances are there are no methods to read the content of the
+ * `FormData` after construction, so the only way to implement `entries`
+ * polyfill is to patch the constructor (and the `append` method).
+ *
+ * For more details on this, see http://mdn.io/FormData.
+ *
+ * @param {!HTMLFormElement=} form An HTML `<form>` element — when specified,
+ *     the FormData object will be populated with the form's current keys/values
+ *     using the name property of each element for the keys and their submitted
+ *     value for the values. It will also encode file input content.
+ * @this {FormData}
+ */
+function FormData(form) {
+  this.map_ = {};
```

Comment:
Minor: We have helper in `utils/object.js#map` that creates prototype-less objects.